File: src_java/Runner.java

import java.io.FileWriter;
import java.io.IOException;
import java.util.InputMismatchException;
import java.util.Scanner;

public class Runner {

    //names of csv files
    static String timesCsvFile = "src_java/times_java.csv";
    static String opsCsvFile = "src_java/ops_java.csv";

    static void initiateFile(String nameOfFile){
        //try to open file, write names of columns into it, then close
        try{
            FileWriter fileWriter =  new FileWriter(nameOfFile);
            fileWriter.write("N,Iterative,Recursive,Dynamic Programming\n");
            fileWriter.close();
        }catch(IOException e){
            System.out.println("Could not write to files");
        }
    }

    public static void main(String[] args){

        //
        boolean repeat = true;
        int N = -1;
        while (N < 0){
            try{
                System.out.println("Enter desired size of table: ");
                Scanner scanner = new Scanner(System.in);
                int input = scanner.nextInt();
                if (input >= 0) N = input;
            }catch (InputMismatchException e){
                System.out.println("Input cannot be negative!");
            }
        }

        //empty files and write names of columns into them
        initiateFile(timesCsvFile);
        initiateFile(opsCsvFile);

        //initiate all fib objects and make array of them
        FibonacciApproach fibonacciIterative = new FibonacciIterative();
        FibonacciApproach fibonacciRecursive = new FibonacciRecursive();
        FibonacciApproach fibonacciRecursiveDynamic = new FibonacciDP(N);

        FibonacciApproach[] fibonaccis = {fibonacciIterative, fibonacciRecursive, fibonacciRecursiveDynamic};


        try{
            //make filewriters in append mode. second arg being true means append mode
            FileWriter timesFileWriter =  new FileWriter(timesCsvFile, true);
            FileWriter opsFileWriter =  new FileWriter(opsCsvFile, true);

            //loop to N
            for (int n = 0; n <= N; n++){
                //append n into file with ,. will have to convert n to int
                opsFileWriter.write(Integer.toString(n) + ",");
                timesFileWriter.write(Integer.toString(n) + ",");

                //loop through fibonacci objects
                for (int i = 0; i < fibonaccis.length; i++){
                    
                    //curr object
                    FibonacciApproach fibonacci = fibonaccis[i];
                    
                    //reset ops
                    fibonacci.setOps(0);
                    
                    //make recursive function not run and print - after certain n number
                    if (i == 1 && n > 50){
                        opsFileWriter.write("-");
                        timesFileWriter.write("-");

                    }else {
                        //run time function passing in object and receive time
                        String time_taken = Helper.time_function(fibonacci, n, 0);
                        
                        //write time and ops into csv files
                        opsFileWriter.write(Integer.toString(fibonacci.getOps()));
                        timesFileWriter.write(time_taken);

                    }
                    //if not running last function, include comma in files
                    if (i != 2) {
                        opsFileWriter.write(",");
                        timesFileWriter.write(",");
                    }

                }
                //go to next line in csv files
                opsFileWriter.write("\n");
                timesFileWriter.write("\n");

            }

            //close files
            opsFileWriter.close();
            timesFileWriter.close();
        }catch(IOException e){
            System.out.println("Could not append to files");
        }

    }
}

File: src_c/runner.c

#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include "fibonacci.h"

void initiateFile(char* nameOfFile){
	//open or create a file
	FILE *fileW;
    fileW = fopen(nameOfFile, "w");
    //overwrite text of files with labels of columns
    fprintf(fileW, "N,Iterative,Recursive,Dynamic Programming\n");
	//close file!
    fclose(fileW);
	
}

int main(int argc, char** argv){

	//if user does not enter an n number, return 1 so that main runs again
    while (argc < 2){
        printf("Enter desired size of table: ");
        return 1;
    }

	//initiate array of functions
    uint64_t (*funcArray[])(uint64_t n, int* ops) = {fibonacciIterative, fibonacciRecursive, fibonacciRecursiveDynamic};
    
	//convert inputted n number from string to int
    int size = atoi(argv[1]);

	//calls initiateFile so that both ops csv and times csv get emptied and get labelled columns
	initiateFile("times.csv");
	initiateFile("ops.csv");
 
	//open file in append mode
    FILE *fileAtime;
    fileAtime = fopen("times.csv", "a");

	//same for ops file
    FILE *fileAops;	
    fileAops = fopen("ops.csv", "a");

    for (int n = 3; n <= size; n++){
	
		//convert n to a string and print it into csv files
    	char nString[100];
    	sprintf(nString, "%d", n);
    	fprintf(fileAops, "%s,", nString);
    	fprintf(fileAtime, "%s,", nString);
		
		//loops through array of functions
        for (int i = 0; i < 3; i++){
	
			int ops = 0;
			
			//make sure recursive function doesn't try to fun after a certain number
			if (i == 1 && n > 50){
				fprintf(fileAtime, "-");
				fprintf(fileAops, "-");
				
			}else {
				//run function and get time taken
				double time_taken = time_function(funcArray[i], (uint64_t)n, 0, &ops);

				//convert time taken to a string and print into csv files
            	char timeString[100];
            	sprintf(timeString, "%.6f", time_taken);
            	fprintf(fileAtime, "%s", timeString);

				//do the same for ops
				char opsString[100];
				sprintf(opsString, "%d", ops);
            	fprintf(fileAops, "%s", opsString);
			}
            
			//if we are not at last function, print a comma to csv files
            if (i != 2) {
                fprintf(fileAtime, ",");
				fprintf(fileAops, ",");
            }   
 
        }
		//after loop, go to next line in csv files
        fprintf(fileAtime, "\n");
		fprintf(fileAops, "\n");
    }

    //close files
    fclose(fileAtime);
	fclose(fileAops);
    
}